go/service/chat_local.go

```
@@ -358,57 +359,64 @@ func (h *chatLocalHandler) getConversationInfo(ctx context.Context, conversation
-			if t, err := body.MessageType(); err != nil {
+			version, err := unboxed.MessagePlaintext.Version()
+			if err != nil {
 				return conversationInfo, maxMessages, err
-			} else if t == chat1.MessageType_METADATA {
-				conversationInfo.TopicName = body.Metadata().ConversationTitle
 			}
-			if unboxed.ServerHeader.MessageID.String() == conversationRemote.ReaderInfo.MaxMsgid.String() {
-				conversationInfo.TlfName = unboxed.MessagePlaintext.V1().ClientHeader.TlfName
+			switch version {
+			case chat1.MessagePlaintextVersion_V1:
+				body := unboxed.MessagePlaintext.V1().MessageBody
+				if t, err := body.MessageType(); err != nil {
+					return conversationInfo, maxMessages, err
+				} else if t == chat1.MessageType_METADATA {
+					conversationInfo.TopicName = body.Metadata().ConversationTitle
+				}
+				if unboxed.ServerHeader.MessageID >= maxValidID {
+					conversationInfo.TlfName = unboxed.MessagePlaintext.V1().ClientHeader.TlfName
+				}
+				conversationInfo.Triple = unboxed.MessagePlaintext.V1().ClientHeader.Conv
+			default:
+				return conversationInfo, maxMessages, libkb.NewChatMessageVersionError(version)
 			}
-			conversationInfo.Triple = unboxed.MessagePlaintext.V1().ClientHeader.Conv
-		default:
-			return conversationInfo, maxMessages, libkb.NewChatMessageVersionError(version)
 		}
-
 	}
 
 	if len(conversationInfo.TlfName) == 0 {
-		return conversationInfo, maxMessages, errors.New("unexpected response from server: global MaxMsgid is not present in MaxHeaders")
+		return conversationInfo, maxMessages, errors.New("no valid message in the conversation")
 	}
 
-	// TODO: verify Conv matches ConversationIDTriple in MessageClientHeader
+	// verify Conv matches ConversationIDTriple in MessageClientHeader
+	if !conversationRemote.Metadata.IdTriple.Eq(conversationInfo.Triple) {
+		return conversationInfo, maxMessages, errors.New("server header conversation triple does not match client header triple")
```

I'm a little worried about returning non-empty conversationInfo, maxMessages when an error occurs.  Here and elsewhere.  If clients ignore the error, they will have invalid data...